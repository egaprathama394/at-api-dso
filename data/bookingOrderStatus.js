export default {
  orderDaimoCode: 'DFA202000000123',
  pssOrderStatus: '2',
};
