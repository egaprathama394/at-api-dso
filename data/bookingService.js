const randomstring = require('randomstring');

const fullNameMax = randomstring.generate({
  charset: 'string',
  length: '110',
});

const noPhoneMax = randomstring.generate({
  charset: 'numeric',
  length: '17',
});

const noPhoneRandom = randomstring.generate({
  charset: 'numeric',
  length: '18',
});

const fullNameRandom = randomstring.generate({
  charset: 'string',
  length: '111',
});

const randomNopol1 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol2 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol3 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol4 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

export default {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AB${randomNopol1}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght

export const maxInput = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameMax}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data booking service dengan inputan max char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneMax}`,
  policeRegNo: `AC${randomNopol2}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght + 1

export const negativeRequest = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameRandom}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service dengan inputan max + 1 char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneRandom}`,
  policeRegNo: `AD${randomNopol3}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for cancel booking service

export const inputDataForCancel = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data untuk cancel booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AA${randomNopol4}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const cancelRequest = {
  orderNo: '',
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

// data for display booking service

export const displayBookingService = {
  isWalkIn: false,
  orderBy: '',
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const invalidDisplayBookingService = {
  isWalkIn: '',
  orderBy: '',
  AccountType: '',
  Action: '',
  ApplicationName: '',
  DeviceType: '',
  OperatingSystem: '',
  UserVersion: '',
};
