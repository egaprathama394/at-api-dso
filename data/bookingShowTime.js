export default {
  branchCode: 'D004',
  scheduleDate: '2020-09-30 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const negativeRequest = {
  branchCode: 'D004',
  scheduleDate: '2018-10-26 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
