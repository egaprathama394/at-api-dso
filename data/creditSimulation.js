export default {
  P_NEW_USED: 'N',
  P_CATEGORY: 'EX2',
  P_VEHICLE_CATEGORY: '1',
  P_BRAND: '002',
  P_TYPE: '1',
  P_MODEL: '1',
  P_DP: '49125000',
  P_OTR: '196500000',
  P_TYPE_INSU: 'A',
  P_MODE_INSU: 'K',
  P_FLAG_ACP: 'Y',
  P_TENORVALUE: '12',
};