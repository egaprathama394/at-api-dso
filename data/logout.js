export default {
  applicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  deviceID: '8B55A4DA-5331-47F9-9C8D-838A110FA18C',
};


export const negativeRequest = {
  action: 'Display',
  applicationName: '9093D285-5263-4B9A-97D7-C185189B0695',
  userVersion: '1.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
