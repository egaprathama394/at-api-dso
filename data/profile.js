export default {
  action: 'Display',
  applicationName: '9093D285-5263-4B9A-97D7-C185189B0695',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  deviceID: '74eef5b5-69c0-454d-9e2d-485c9e988766',
};

export const editProfile = {
  userAccount: '',
  AccountType: '1',
  Action: 'Edit',
  ApplicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};
