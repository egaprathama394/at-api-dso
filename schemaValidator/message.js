export default {
  type: 'object',
  required: [
    'messages',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    messages: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'messageParentId',
          'userId',
          'categoryId',
          'isRead',
          'isClosed',
          'title',
          'message',
          'branchCode',
          'branchName',
          'name',
          'address',
          'phone',
          'sendFrom',
          'lastPostId',
          'isClientDeleted',
          'isDealerDeleted',
          'message1',
          'id',
          'totalMessage',
          'customerCare',
          'createdBy',
          'createdTime',
          'modifiedBy',
          'modifiedTime',
        ],
        properties: {
          messageParentId: {
            type: 'integer',
          },
          userId: {
            type: 'string',
          },
          categoryId: {
            type: 'integer',
          },
          isRead: {
            type: 'boolean',
          },
          isClosed: {
            type: 'boolean',
          },
          title: {
            type: 'string',
          },
          message: {
            type: 'string',
          },
          branchCode: {
            type: 'string',
          },
          branchName: {
            type: 'string',
          },
          name: {
            type: 'string',
          },
          address: {
            type: 'string',
          },
          phone: {
            type: 'string',
          },
          sendFrom: {
            type: 'string',
          },
          lastPostId: {
            type: 'integer',
          },
          isClientDeleted: {
            type: 'boolean',
          },
          isDealerDeleted: {
            type: 'boolean',
          },
          message1: {
            type: 'string',
          },
          id: {
            type: 'integer',
          },
          totalMessage: {
            type: 'integer',
          },
          customerCare: {
            type: 'string',
          },
          createdBy: {
            type: 'string',
          },
          createdTime: {
            type: 'string',
          },
          modifiedBy: {
            type: 'string',
          },
          modifiedTime: {
            type: 'string',
          },
        },
      },
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

// schema for autoread messages

export const autoreadSchema = {
  type: 'object',
  required: [
    'messages',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    messages: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'messageParentId',
          'userId',
          'categoryId',
          'isRead',
          'isClosed',
          'title',
          'message',
          'branchCode',
          'branchName',
          'name',
          'address',
          'phone',
          'sendFrom',
          'lastPostId',
          'isClientDeleted',
          'isDealerDeleted',
          'message1',
          'id',
          'totalMessage',
          'createdBy',
          'createdTime',
          'modifiedBy',
          'modifiedTime',
        ],
        properties: {
          messageParentId: {
            type: 'integer',
          },
          userId: {
            type: 'string',
          },
          categoryId: {
            type: 'integer',
          },
          isRead: {
            type: 'boolean',
          },
          isClosed: {
            type: 'boolean',
          },
          title: {
            type: 'string',
          },
          message: {
            type: 'string',
          },
          branchCode: {
            type: 'string',
          },
          branchName: {
            type: 'string',
          },
          name: {
            type: 'string',
          },
          address: {
            type: 'string',
          },
          phone: {
            type: 'string',
          },
          sendFrom: {
            type: 'string',
          },
          lastPostId: {
            type: 'integer',
          },
          isClientDeleted: {
            type: 'boolean',
          },
          isDealerDeleted: {
            type: 'boolean',
          },
          message1: {
            type: 'string',
          },
          id: {
            type: 'integer',
          },
          totalMessage: {
            type: 'integer',
          },
          createdBy: {
            type: 'string',
          },
          createdTime: {
            type: 'string',
          },
          modifiedBy: {
            type: 'string',
          },
          modifiedTime: {
            type: 'string',
          },
        },
      },
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

// schema for negative request autoread message test

export const negativerequest = {
  type: 'object',
  required: [
    'acknowledge',
    'message',
    'version',
    'build',
  ],
  properties: {
    acknowledge: {
      type: 'integer',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

// schema for success post messages

export const postMessages = {
  type: 'object',
  required: [
    'acknowledge',
    'message',
    'version',
    'build',
  ],
  properties: {
    acknowledge: {
      type: 'integer',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const negativeRequest = {
  type: 'object',
  properties: {
    message: {
      type: 'string',
    },
    modelState: {
      type: 'object',
      properties: {
        'userId.String': {
          type: 'array',
          items: {
            type: 'string',
          },
        },
      },
    },
  },
};
