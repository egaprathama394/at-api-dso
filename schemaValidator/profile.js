export default {
  properties: {
    userAccount: {
      type: 'object',
      properties: {
        email: {
          type: 'string',
        },
        firstName: {
          type: 'string',
        },
        lastName: {
          type: 'string',
        },
        phoneNo: {
          type: 'string',
        },
        address: {
          type: 'string',
        },
        provinceCode: {
          type: 'string',
        },
        districtCode: {
          type: 'string',
        },
        subdistrictCode: {
          type: 'string',
        },
        villageCode: {
          type: 'string',
        },
        zipCode: {
          type: 'string',
        },
        birthdate: {
          type: 'string',
        },
        gender: {
          type: 'integer',
        },
        religion: {
          type: 'integer',
        },
        hobby: {
          type: 'string',
        },
        location: {
          type: 'string',
        },
        identificationType: {
          type: 'string',
        },
        identificationNumber: {
          type: 'string',
        },
        identificationDate: {
          type: 'string',
        },
        urlPhotoProfile: {
          type: 'string',
        },
        occupation: {
          type: 'string',
        },
        defaultPaymentMethod: {
          type: 'integer',
        },
        referralCode: {
          type: 'string',
        },
      },
      required: [
        'email',
        'firstName',
        'lastName',
        'phoneNo',
        'address',
        'provinceCode',
        'districtCode',
        'subdistrictCode',
        'villageCode',
        'zipCode',
        'birthdate',
        'gender',
        'religion',
        'hobby',
        'location',
        'identificationType',
        'identificationNumber',
        'identificationDate',
        'urlPhotoProfile',
        'occupation',
        'defaultPaymentMethod',
        'referralCode',
      ],
    },
    acknowledge: {
      type: 'integer',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'number',
    },

  },
  required: [
    'userAccount',
    'acknowledge',
    'message',
    'version',
    'build',
  ],
  additionalProperties: false,
};

export const invalidToken = {
  properties: {
    error: {
      type: 'string',
    },
    error_description: {
      type: 'string',
    },
  },
  required: [
    'error',
    'error_description',
  ],
  additionalProperties: false,

};

export const editProfileSchema = {
  properties: {
    acknowledge: {
      type: 'number',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'number',
    },
  },
  required: [
    'acknowledge',
    'message',
    'version',
    'build',
    'errorCode',
  ],
  additionalProperties: false,
};