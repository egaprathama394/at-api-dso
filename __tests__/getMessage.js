import getMessage from '../endpoints/message';
import * as messageSchema from '../schemaValidator/message';
import getProfile from '../endpoints/profile';
import * as profileSchema from '../schemaValidator/profile';
import * as body from '../data/profile';

const fs = require('fs');
const { matchers } = require('jest-json-schema');

expect.extend(matchers);

const ambilToken = fs.readFileSync('./token.json');
const objToken = JSON.parse(ambilToken);

const token = objToken.login;

describe('Get Message Contact Us', () => {
  test('Can get all message at contact us successfully using authorized token - positive test cases', async () => {
    const response = await getProfile(token, body.default);
    expect(response.status).toEqual(200);
    expect(response.body).toMatchSchema(profileSchema.default);
    const userId = await response.body.userAccount.email;
    const queryParam = {
      userId: `${userId}`,
    };
    const responseMessage = await getMessage(token, queryParam);
    expect(responseMessage.status).toEqual(200);
    expect(responseMessage.body).toMatchSchema(messageSchema.default);
    Reflect.deleteProperty(responseMessage.body.messages[0], 'modifiedTime');
    expect(responseMessage.body.messages[0]).toMatchSnapshot();
  });

  test('Can not get message using invalid userId - negative test cases', async () => {
    const queryParam = {
      userId: 'xxxyyy@gmail.com',
    };
    const responseMessage = await getMessage(token, queryParam);
    expect(responseMessage.status).toEqual(200);
    expect(responseMessage.body).toMatchSchema(messageSchema.default);
    expect(responseMessage.body).toMatchSnapshot();
  });

  test('Can not get message using empty userId - negative test cases', async () => {
    const queryParam = {
      userId: '',
    };
    const responseMessage = await getMessage(token, queryParam);
    expect(responseMessage.status).toEqual(400);
    expect(responseMessage.body).toMatchSchema(messageSchema.negativeRequest);
    expect(responseMessage.body).toMatchSnapshot();
  });

  test('Can not get message using null userId - negative test cases', async () => {
    const queryParam = {
      userId: null,
    };
    const responseMessage = await getMessage(token, queryParam);
    expect(responseMessage.status).toEqual(400);
    expect(responseMessage.body).toMatchSchema(messageSchema.negativeRequest);
    expect(responseMessage.body).toMatchSnapshot();
  });

  test('Can not get message without queryParam - negative test cases', async () => {
    const responseMessage = await getMessage(token);
    expect(responseMessage.status).toEqual(200);
    expect(responseMessage.body).toMatchSchema(messageSchema.default);
    expect(responseMessage.body).toMatchSnapshot();
  });
});
