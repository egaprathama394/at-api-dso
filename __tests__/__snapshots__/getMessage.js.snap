// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Message Contact Us Can get all message at contact us successfully using authorized token - positive test cases 1`] = `
Object {
  "address": "",
  "branchCode": "HO",
  "branchName": "Head Office",
  "categoryId": 108,
  "createdBy": "testerlogin.DSO@gmail.com",
  "createdTime": "2019-10-29 14:32:08",
  "customerCare": "",
  "id": 16562,
  "isClientDeleted": false,
  "isClosed": false,
  "isDealerDeleted": false,
  "isRead": true,
  "lastPostId": 0,
  "message": "halo halo bandung",
  "message1": "halo halo bandung",
  "messageParentId": 0,
  "modifiedBy": "testerlogin.DSO@gmail.com",
  "name": "tester login",
  "phone": "081903150019",
  "sendFrom": "tester login",
  "title": "halohalo",
  "totalMessage": 1,
  "userId": "testerlogin.DSO@gmail.com",
}
`;

exports[`Get Message Contact Us Can not get message using empty userId - negative test cases 1`] = `
Object {
  "message": "The request is invalid.",
  "modelState": Object {
    "userId.String": Array [
      "A value is required but was not present in the request.",
    ],
  },
}
`;

exports[`Get Message Contact Us Can not get message using invalid userId - negative test cases 1`] = `
Object {
  "acknowledge": 1,
  "build": "0",
  "messages": Array [],
  "version": "1.1",
}
`;

exports[`Get Message Contact Us Can not get message using null userId - negative test cases 1`] = `
Object {
  "message": "The request is invalid.",
  "modelState": Object {
    "userId.String": Array [
      "A value is required but was not present in the request.",
    ],
  },
}
`;

exports[`Get Message Contact Us Can not get message without queryParam - negative test cases 1`] = `
Object {
  "acknowledge": 0,
  "build": "0",
  "messages": Array [],
  "version": "1.1",
}
`;
